### PR TITLE
chore: pulish playwright reports using github action (v2)

### PR DESCRIPTION
**Pull request by mas-who**
## Done

- Publish playwright reports using github actions using `workflow_run` event trigger

### CODE REVIEW

File: .github/workflows/publish_test_reports.yml [hunk 1]
```diff
@@ -0,0 +1,82 @@
...
+    publish_report:
+        name: publish-e2e-report
+        runs-on: ubuntu-latest
+        continue-on-error: true
+        steps:
+          - uses: actions/checkout@v4
+            with:
+              ref: gh-pages
+              token: ${{ secrets.GITHUB_TOKEN }}
+          - name: Dlownload test report artifact
```
Comment (edlerd) [inline .github/workflows/publish_test_reports.yml:17]:
```suggestion
          - name: Download test report artifact
```

**Review by edlerd — APPROVED** [review]
LGTM

We could simplify further -- maybe no need to parse the pr number, run id and run attempt for the commit message. The commit message could be the same each time, or just contain the full path as it is. But also fine to keep as it is from my side.

**Comment by mas-who** [comment]
> LGTM
> 
> We could simplify further -- maybe no need to parse the pr number, run id and run attempt for the commit message. The commit message could be the same each time, or just contain the full path as it is. But also fine to keep as it is from my side.

Oh my bad, I misunderstood you yesterday, thought you wanted to not store the additional information in the info.txt but still display them. Will just display the path in the commit message then.

File: .github/workflows/publish_test_reports.yml [hunk 2]
```diff
@@ -0,0 +1,79 @@
...
+          # actual authentication is done using secrets.GITHUB_TOKEN with write permission
+          - name: Set Git User
+            run: |
+              git config --global user.email "github-action@example.com"
+              git config --global user.name "GitHub Action"
+          - name: Push HTML Report
+            timeout-minutes: 3
+            run: |
+              git add .
+              git commit -m "workflow: add HTML report publised at $HTML_REPORT_URL_PATH"
```
Comment (edlerd) [inline .github/workflows/publish_test_reports.yml:60]:
```suggestion
              git commit -m "workflow: add HTML report published at $HTML_REPORT_URL_PATH"
```

File: .github/workflows/delete_test_reports.yml [hunk 3]
```diff
@@ -0,0 +1,52 @@
...
+                        echo "Failed to rebase. Please review manually."
+                        exit 1
+                    fi
+        
+                    git push
+                    if [ $? -eq 0 ]; then
+                        echo "Successfully pushed HTML reports to repo."
+                        exit 0
+                    fi
+                done
```
Comment (edlerd) [inline .github/workflows/delete_test_reports.yml:52]:
```suggestion
                done

```

**Review by edlerd — APPROVED** [review]
LGTM, thanks for the changes. Two minor typos below, then ready for merge :+1:

**Pull request merged**